daffodil-cli/src/main/scala/org/apache/daffodil/Main.scala

```diff
@@ -592,7 +592,9 @@ object Main {
     try {
       val compiler = Compiler()
       val processor = Timer.getResult("reloading", compiler.reload(savedParser))
-      displayDiagnostics(processor)
+      // note that we do not display the diagnostics that were saved as part of the
+      // saved processor. Those are from compile time, are all warnings, and
+      // are just noise in a production system where we're reloading a parser.
```

Comment:
Just to confirm, we do actually keep compile time and runtime diagnostics completely separate and compile time warnings do not bleed into the ParseResult diagnostics? And I guess the TDML runner combines them both when checking for expected warnings when a test finishes? That's not how I thought it worked, but good to know I was wrong because that seems like a logical way for it to work.

Reply:
There's one heap of diagnostics held in static state on the SchemaSetRuntimeData object. This is compilation-time warnings only.
Runtime diagnostics are gathered in the PState/UState. They have to be because of backtracking (well for parsing anyway) we can't be mutating the diagnostics in the SSRD at runtime since it is shared cross threads.